DirectProgramming/DPC++FPGA/ReferenceDesigns/gzip/src/gzipkernel_ll.cpp

```diff
@@ -53,12 +53,13 @@
 using namespace sycl;
 
 // Pipes, for inter-kernel data transfer..
-using acc_dist_channel_array = PipeArray<  // Defined in "pipe_utils.hpp".
-    class dist_channel_pipe_id,            // An identifier for the pipe.
-    struct DistLen,                        // The type of data in the pipe.
-    32,                                     // The capacity of each pipe.
-    NUM_ENGINES                            // array dimension.
-    >;
+using acc_dist_channel_array =
+  fpga_tools::PipeArray<          // Defined in "pipe_utils.hpp".
+    class dist_channel_pipe_id,   // An identifier for the pipe.
+    struct DistLen,               // The type of data in the pipe.
```

Comment:
Curious mix of `class` and `struct` here.

Reply:
The `struct` keyword is unnecessary here, but I think the use of a struct is fine: `DistLen` is just a bundle of a couple of types.